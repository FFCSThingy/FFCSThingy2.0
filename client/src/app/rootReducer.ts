import { combineReducers } from 'redux';

import miscReducer,
{
	initialState as miscInitState,
	name as miscName,
} from '../reducers/misc';

import courseReducer,
{
	initialState as courseInitState,
	name as courseName,
} from '../reducers/course';

import timetableReducer,
{
	initialState as timetableInitState,
	name as timetableName,
} from '../reducers/timetable';

import curriculumReducer,
{
	initialState as curriculumInitState,
	name as curriculumName,
} from '../reducers/curriculum';

import userReducer,
{
	initialState as userInitState,
	name as userName,
} from '../reducers/user';

import authReducer,
{
	initialState as authInitState,
	name as authName,
} from '../reducers/auth';

const rootReducer = combineReducers({
	misc: miscReducer,
	course: courseReducer,
	timetable: timetableReducer,
	curriculum: curriculumReducer,
	user: userReducer,
	auth: authReducer,
});

export const initialState = {
	[miscName]: miscInitState,
	[courseName]: courseInitState,
	[timetableName]: timetableInitState,
	[curriculumName]: curriculumInitState,
	[userName]: userInitState,
	[authName]: authInitState,
};

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
