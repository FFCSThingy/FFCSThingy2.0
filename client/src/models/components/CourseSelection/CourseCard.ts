import CourseReqs from '../../data/CourseReqs';

export default interface CourseCardProps {
	code: string;
	title: string;
	credits: number;
	shortCourseTypes: string[];
	completed?: string;
	selected?: boolean;
	prereqs?: CourseReqs;
}

export interface PrereqTextProps {
	value: string;
}
