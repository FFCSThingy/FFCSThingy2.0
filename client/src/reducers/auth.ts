import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit';
import AuthSlice from '../models/state/AuthSlice';
import { getUserDetails, doLogout } from '../api/user';

const ACTION_BASE = 'auth';

export const checkAuth = createAsyncThunk(
	`${ACTION_BASE}/checkAuth`,
	async () => {
		const data = await getUserDetails();
		return data;
	},
);

export const logout = createAsyncThunk(
	`${ACTION_BASE}/logout`,
	async () => {
		const data = await doLogout();
		return data;
	},
);

export const initialState: AuthSlice = {
	isAuthenticated: null,
};

const authSlice = createSlice({
	name: ACTION_BASE,
	initialState,
	reducers: {
		changeAuth: (state, action: PayloadAction<boolean>) => {
			state.isAuthenticated = action.payload;
		},
	},
	extraReducers: (builder) => {
		builder.addCase(
			checkAuth.fulfilled,
			(state) => {
				state.isAuthenticated = true;
			},
		).addCase(
			checkAuth.rejected,
			(state) => {
				state.isAuthenticated = false;
			},
		);

		builder.addCase(
			logout.fulfilled,
			(state) => {
				state.isAuthenticated = false;
			},
		).addCase(
			logout.rejected,
			(state) => {
				state.isAuthenticated = false;
			},
		);
	},
});

export const { changeAuth } = authSlice.actions;
export const { name } = authSlice;
export default authSlice.reducer;
