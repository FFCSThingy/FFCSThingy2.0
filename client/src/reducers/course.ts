import {
	createSlice,
	createAsyncThunk,
	PayloadAction,
} from '@reduxjs/toolkit';

import { getHeatmap, getAllCourseLists } from '../api/course';

import HeatmapCourse from '../models/data/HeatmapCourse';
import {
	CourseList,
	CourseSlotList,
	CourseTypeList,
	CourseFacultyList,
} from '../models/data/CourseLists';
import RequisitesList from '../models/data/RequisitesList';
import CourseSlice from '../models/state/CourseSlice';

const ACTION_BASE = 'course';

export const fetchHeatmap = createAsyncThunk(
	`${ACTION_BASE}/fetchHeatmap`,
	async (timestamp: string) => {
		const data = await getHeatmap(timestamp);
		return data;
	},
);

export const fetchAllCourseLists = createAsyncThunk(
	`${ACTION_BASE}/fetchAllCourseLists`,
	async (timestamp: string) => {
		const data = await getAllCourseLists(timestamp);
		return data;
	},
);

export const initialState: CourseSlice = {
	selected: '',
	heatmap: {
		data: [],
		timestamp: '',
		syncing: false,
	},
	lists: {
		course: {},
		slot: {},
		faculty: {},
		type: {},
		req: {},
		timestamp: '',
	},
};

const courseSlice = createSlice({
	name: ACTION_BASE,
	initialState,
	reducers: {
		selectCourse: (state, action: PayloadAction<string>) => {
			state.selected = action.payload;
		},
	},
	extraReducers: (builder) => {
		builder.addCase(
			fetchHeatmap.pending,
			(state) => {
				state.heatmap.syncing = true;
			},
		).addCase(
			fetchHeatmap.rejected,
			(state) => {
				state.heatmap.syncing = false;
			},
		).addCase(
			fetchHeatmap.fulfilled,
			(
				state,
				action: PayloadAction<{
					heatmap: HeatmapCourse[],
					timestamp: string
				}>,
			) => {
				const { heatmap, timestamp } = action.payload;

				state.heatmap.data = heatmap;
				state.heatmap.timestamp = timestamp;
				state.heatmap.syncing = false;
			},
		);

		builder.addCase(
			fetchAllCourseLists.fulfilled,
			(state, action: PayloadAction<{
				courseList: CourseList,
				courseSlotList: CourseSlotList,
				courseFacultyList: CourseFacultyList,
				courseTypeList: CourseTypeList,
				prerequisites: RequisitesList,
				timestamp: string
			}>) => {
				const {
					courseList,
					courseSlotList,
					courseFacultyList,
					courseTypeList,
					prerequisites,
					timestamp,
				} = action.payload;

				state.lists.course = courseList;
				state.lists.slot = courseSlotList;
				state.lists.faculty = courseFacultyList;
				state.lists.type = courseTypeList;
				state.lists.req = prerequisites;
				state.lists.timestamp = timestamp;
			},
		);
	},
});

export const {
	selectCourse,
} = courseSlice.actions;
export const { name } = courseSlice;
export default courseSlice.reducer;
