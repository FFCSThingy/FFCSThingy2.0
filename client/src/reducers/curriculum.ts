import {
	createSlice,
	PayloadAction,
	createAsyncThunk,
} from '@reduxjs/toolkit';

import { getCurriculumPrefixes, getCurriculumByPrefix } from '../api/curriculum';

import Curriculum from '../models/data/Curriculum';
import CurriculumSlice from '../models/state/CurriculumSlice';

const ACTION_BASE = 'curriculum';

export const fetchCurriculumPrefixes = createAsyncThunk(
	`${ACTION_BASE}/fetchCurriculumPrefixes`,
	async () => {
		const data = await getCurriculumPrefixes();
		return data;
	},
);

export const fetchCurriculumFromPrefix = createAsyncThunk(
	`${ACTION_BASE}/fetchCurriculumFromPrefix`,
	async (prefix: string) => {
		const data = await getCurriculumByPrefix(prefix);
		return data;
	},
);

export const initialState: CurriculumSlice = {
	selectedPrefix: 'Curriculum',
	list: [] as string[],
	currentData: {
		_id: '',
		reg_prefix: '',
		__v: 0,
		todo_creds: {
			pc: 0,
			uc: 0,
			pe: 0,
			ue: 0,
		},
		bridge: [],
		pc: [],
		uc: [],
		pe: [],
		ue: [],
	},
};

const curriculumSlice = createSlice({
	name: ACTION_BASE,
	initialState: { ...initialState },
	reducers: {
		setPrefixList: {
			prepare(list: string[]) {
				return { payload: { list } };
			},
			reducer(state, action: PayloadAction<{ list: string[] }, string>) {
				const { list } = action.payload;
				state.list = list;
			},
		},
		setSelectedCurriculum: {
			prepare(selectedPrefix: string) {
				return { payload: { selectedPrefix } };
			},
			reducer(state, action: PayloadAction<{ selectedPrefix: string }, string>) {
				const { selectedPrefix } = action.payload;
				state.selectedPrefix = selectedPrefix;
			},
		},
		setCurrentCurriculumData: {
			prepare(currentCurriculum: Curriculum) {
				return { payload: { currentCurriculum } };
			},
			reducer(state, action: PayloadAction<{ currentCurriculum: Curriculum }, string>) {
				const { currentCurriculum } = action.payload;
				state.currentData = currentCurriculum;
			},
		},
	},
	extraReducers: (builder) => {
		builder.addCase(
			fetchCurriculumPrefixes.fulfilled,
			(state, action: PayloadAction<string[]>) => {
				const list = action.payload;
				if (
					!list.includes(state.selectedPrefix)
					&& state.selectedPrefix !== 'Curriculum'
				) {
					[state.selectedPrefix] = list;
				}

				state.list = list;
			},
		);

		builder.addCase(
			fetchCurriculumFromPrefix.fulfilled,
			(state, action: PayloadAction<Curriculum>) => {
				const data = action.payload;
				state.currentData = data;
			},
		).addCase(
			fetchCurriculumFromPrefix.rejected,
			(state) => {
				state.selectedPrefix = 'Curriculum';
				state.currentData = {
					_id: '',
					reg_prefix: '',
					__v: 0,
					todo_creds: {
						pc: 0,
						uc: 0,
						pe: 0,
						ue: 0,
					},
					bridge: [],
					pc: [],
					uc: [],
					pe: [],
					ue: [],
				};
			},
		);
	},
});

export const {
	setPrefixList,
	setSelectedCurriculum,
	setCurrentCurriculumData,
} = curriculumSlice.actions;

export const { name } = curriculumSlice;
export default curriculumSlice.reducer;
