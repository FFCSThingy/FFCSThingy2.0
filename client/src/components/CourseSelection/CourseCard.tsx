import React, { memo, FC } from 'react';
import { Card, OverlayTrigger, Tooltip } from 'react-bootstrap';
import { useDispatch } from 'react-redux';

import { selectCourse } from '../../reducers/course';

import styles from '../../css/CourseSelectionList.module.scss';

import CourseCardProps,
{ PrereqTextProps } from '../../models/components/CourseSelection/CourseCard';

const CourseCard: FC<CourseCardProps> = memo(
	({
		code, title, credits, shortCourseTypes,
		completed = '', selected = false, prereqs = null,
	}) => {
		const dispatch = useDispatch();

		const cardClass = selected
			? `${styles.courseCard} ${styles.active}`
			: styles.courseCard;

		const completionClass = ['N', 'F'].includes(completed)
			? styles.cardCompletionFailedSubtitle
			: styles.cardCompletedSubtitle;

		const creditText = (credits === 1) ? 'Credit' : 'Credits';

		const valueMap = (value: string) => value.split(',').map((v) => (
			<React.Fragment key={`${v}`}>
				{v}
				<br />
			</React.Fragment>
		));

		const CoreqText: FC<PrereqTextProps> = ({ value }) => {
			if (value) {
				return (
					<span key={`${code}-Coreq`}>
						<b>Co-Requisites: </b>
						{valueMap(value)}
						<br />
					</span>
				);
			}
			return <></>;
		};

		const PrereqText: FC<PrereqTextProps> = ({ value }) => {
			if (value) {
				return (
					<span key={`${code}-Prereq`}>
						<b>Pre-Requisites: </b>
						{valueMap(value)}
						<br />
					</span>
				);
			}
			return <></>;
		};
		const AntireqText: FC<PrereqTextProps> = ({ value }) => {
			if (value) {
				return (
					<span key={`${code}-Antireq`}>
						<b>Anti-Requisites: </b>
						{valueMap(value)}
						<br />
					</span>
				);
			}
			return <></>;
		};

		const PrereqsTooltip = () => {
			if (prereqs) {
				return (
					<div className={styles.coursePrereq}>
						<PrereqText value={prereqs.prereq || ''} />
						<CoreqText value={prereqs.coreq || ''} />
						<AntireqText value={prereqs.antireq || ''} />
					</div>
				);
			}
			return <></>;
		};

		const handleClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
			dispatch(selectCourse(e.currentTarget.title));
		};

		return (
			<OverlayTrigger
				key={`${code}-Prereq-Overlay`}
				placement="top-start"
				trigger={['hover', 'focus']}
				overlay={(
					<Tooltip id={`${code}-PrereqsTooltip`} role="tooltip">
						<PrereqsTooltip />
						<div className={styles.courseTypes}>
							<b> Course Type: </b>
							{shortCourseTypes.join(' | ')}
						</div>
					</Tooltip>
				)}
			>
				<Card
					className={cardClass}
					onClick={handleClick}
					title={code}	// For using with handleClick
				>
					<Card.Body>
						<Card.Title className={styles.cardTitle}>
							{title}
						</Card.Title>

						<Card.Text className={styles.courseSelectDetails}>
							<span className={styles.courseCodeText}>
								{code}
							</span>

							<span className={styles.courseCredits}>
								{`${credits} ${creditText}`}
							</span>
						</Card.Text>

						<Card.Subtitle className={completionClass}>
							{
								(completed)
									? (`Completed: ${completed}`) : ''
							}
						</Card.Subtitle>
					</Card.Body>
				</Card>
			</OverlayTrigger>
		);
	},
);

export default CourseCard;
