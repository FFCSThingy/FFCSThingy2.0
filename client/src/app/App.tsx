import React, { useEffect } from 'react';
import {
	BrowserRouter as Router, Switch, Route, Redirect,
} from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';

import ReactGA from 'react-ga';

import Login from '../components/Login';
import Dashboard from '../components/Dashboard';

import { checkAuth } from '../reducers/auth';
import { selectIsAuthenticated } from '../selectors/auth';

ReactGA.initialize('UA-156974674-1');
ReactGA.pageview(window.location.pathname + window.location.search);

const App = () => {
	const dispatch = useDispatch();
	const isAuthenticated = useSelector(selectIsAuthenticated);

	useEffect(() => {
		dispatch(checkAuth());
	// Don't add more dependencies
	// Only needs to run once
	// eslint-disable-next-line react-hooks/exhaustive-deps
	}, []);

	return (
		<Router>
			<Switch>
				<Route path="/login">
					{!isAuthenticated
						? <Login />
						: <Redirect to="/" />}
				</Route>

				<Route path="/">
					<Dashboard />
					{/* {isAuthenticated
						? <Dashboard />
						: <Redirect to="/login" />} */}
				</Route>
			</Switch>
		</Router>
	);
};

export default App;
