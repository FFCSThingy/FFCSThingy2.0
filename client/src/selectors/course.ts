import { RootState } from '../app/rootReducer';

export const selectHeatmapTimestamp = (state: RootState) => state.course.heatmap.timestamp;
export const selectListsTimestamp = (state: RootState) => state.course.lists.timestamp;
export const selectSelectedCourse = (state: RootState) => state.course.selected;
export const selectPrereqList = (state: RootState) => state.course.lists.req;
export const selectHeatmap = (state: RootState) => state.course.heatmap.data;
