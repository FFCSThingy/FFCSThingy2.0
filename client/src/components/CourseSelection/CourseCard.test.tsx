import React from 'react';
import {
	render, fireEvent, screen,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import CourseCard from './CourseCard';

import { initialState } from '../../app/rootReducer';
import { selectCourse } from '../../reducers/course';

const mockStore = configureStore([thunk]);
const store = mockStore(initialState);

const props = {
	code: 'ZZZ1001',
	title: 'Sample1',
	credits: 4,
	shortCourseTypes: ['L', 'T', 'P'],
	selected: false,
	prereqs: {},
};

const samplePrereqs = {
	coreq: 'ZZZ1002',
	antireq: 'ZZZ1003',
	prereq: 'ZZZ1004',
};

describe('component: CourseCard', () => {
	afterEach(() => {
		store.clearActions();
	});

	describe('snapshots', () => {
		it('not-selected, no-prereq, no-completed, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('selected-not-passed, no-prereq, no-completed, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('not-passed, prereq-not-passed, no-completed, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('selected, no-prereq, no-completed, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('not-selected, no-prereq, no-completed, 1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={1}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('not-selected, no-prereq, no-completed, >1 credit, 1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={['L']}
						selected={props.selected}
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('not-selected, no-prereq, completed-pass, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={props.prereqs}
						completed="A"
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('not-selected, no-prereq, completed-fail, >1 credit, >1 courseType', () => {
			expect.hasAssertions();
			const { asFragment } = render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={props.prereqs}
						completed="F"
					/>
				</Provider>,
			);

			expect(asFragment()).toMatchSnapshot();
		});

		it('should show prereq tooltip if prereqs given', () => {
			expect.hasAssertions();
			render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={samplePrereqs}
					/>
				</Provider>,
			);

			fireEvent.mouseOver(screen.getByTitle(props.code));

			expect(screen.getByRole('tooltip'))
				.toMatchSnapshot();
		});

		it('should show only course types if no prereqs given', () => {
			expect.hasAssertions();
			render(
				<Provider store={store}>
					<CourseCard
						code={props.code}
						title={props.title}
						credits={props.credits}
						shortCourseTypes={props.shortCourseTypes}
						selected={props.selected}
						prereqs={props.prereqs}
					/>
				</Provider>,
			);

			fireEvent.mouseOver(screen.getByTitle(props.code));

			expect(screen.getByRole('tooltip'))
				.toMatchSnapshot();
		});
	});

	it(`should dispatch ${selectCourse} on click`, () => {
		expect.hasAssertions();
		render(
			<Provider store={store}>
				<CourseCard
					code={props.code}
					title={props.title}
					credits={props.credits}
					shortCourseTypes={props.shortCourseTypes}
					selected={props.selected}
					prereqs={props.prereqs}
				/>
			</Provider>,
		);

		fireEvent.click(screen.getByTitle(props.code));

		const actions = store.getActions();
		const selectAction = actions.find((v) => v.type === selectCourse.type);
		expect(selectAction.payload).toStrictEqual(props.code);
	});
});
