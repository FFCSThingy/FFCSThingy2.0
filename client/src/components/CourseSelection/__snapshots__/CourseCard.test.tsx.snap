// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CourseCard snapshots not-passed, prereq-not-passed, no-completed, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots not-selected, no-prereq, completed-fail, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletionFailedSubtitle card-subtitle h6"
      >
        Completed: F
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots not-selected, no-prereq, completed-pass, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      >
        Completed: A
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots not-selected, no-prereq, no-completed, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots not-selected, no-prereq, no-completed, >1 credit, 1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots not-selected, no-prereq, no-completed, 1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          1 Credit
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots selected, no-prereq, no-completed, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard active card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots selected-not-passed, no-prereq, no-completed, >1 credit, >1 courseType 1`] = `
<DocumentFragment>
  <div
    class="courseCard card"
    title="ZZZ1001"
  >
    <div
      class="card-body"
    >
      <div
        class="cardTitle card-title h5"
      >
        Sample1
      </div>
      <p
        class="courseSelectDetails card-text"
      >
        <span
          class="courseCodeText"
        >
          ZZZ1001
        </span>
        <span
          class="courseCredits"
        >
          4 Credits
        </span>
      </p>
      <div
        class="cardCompletedSubtitle card-subtitle h6"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`component: CourseCard snapshots should show only course types if no prereqs given 1`] = `
<div
  class="fade show tooltip bs-tooltip-top-start"
  id="ZZZ1001-PrereqsTooltip"
  role="tooltip"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
  x-placement="top-start"
>
  <div
    class="arrow"
  />
  <div
    class="tooltip-inner"
  >
    <div
      class="coursePrereq"
    />
    <div
      class="courseTypes"
    >
      <b>
         Course Type: 
      </b>
      L | T | P
    </div>
  </div>
</div>
`;

exports[`component: CourseCard snapshots should show prereq tooltip if prereqs given 1`] = `
<div
  class="fade show tooltip bs-tooltip-top-start"
  id="ZZZ1001-PrereqsTooltip"
  role="tooltip"
  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
  x-placement="top-start"
>
  <div
    class="arrow"
  />
  <div
    class="tooltip-inner"
  >
    <div
      class="coursePrereq"
    >
      <span>
        <b>
          Pre-Requisites: 
        </b>
        ZZZ1004
        <br />
        <br />
      </span>
      <span>
        <b>
          Co-Requisites: 
        </b>
        ZZZ1002
        <br />
        <br />
      </span>
      <span>
        <b>
          Anti-Requisites: 
        </b>
        ZZZ1003
        <br />
        <br />
      </span>
    </div>
    <div
      class="courseTypes"
    >
      <b>
         Course Type: 
      </b>
      L | T | P
    </div>
  </div>
</div>
`;
